src/pybind/mgr/dashboard/module.py

```diff
@@ -3,6 +3,18 @@
 ceph dashboard mgr plugin (based on CherryPy)
 """
 from __future__ import absolute_import
+from .plugins import PLUGIN_MANAGER
+from .settings import options_command_list, options_schema_list, \
+    handle_option_command
+from .services.exception import dashboard_exception_handler
+from .services.sso import SSO_COMMANDS, \
+    handle_sso_command
+from .services.auth import AuthManager, AuthManagerTool, JwtManager
+from .tools import NotificationQueue, RequestLoggingTool, TaskManager, \
+    prepare_url_prefix, str_to_bool
+from .grafana import push_local_dashboards
+from .controllers import generate_routes, json_error_page
+from . import mgr
```

Comment:
This is really surprising to me, based on [PEP-8](https://www.python.org/dev/peps/pep-0008/#imports):

> Imports should be grouped in the following order:
> 
> 1. Standard library imports.
> 2. Related third party imports.
> 3. Local application/library specific imports.
>
> You should put a blank line between each group of imports.

Perhaps we have different rules enabled/disabled in the linter and in the fixer?

Reply:
> Perhaps we have different rules enabled/disabled in the linter and in the fixer?

I have noticed this before, linter will output no errors, but fix will change a lot of files.
I think we should include in this PR changes to the lint/fix rules so they are equal, otherwise this issues will happen again in the next PRs.

Reply:
+1 for making linter and fix rule be identical. Otherwise, this will always happen.

Reply:
> Perhaps we have different rules enabled/disabled in the linter and in the fixer?

`autopep8` which is used in the `fix` tox environment does not have any configuration (except some really low level settings like line length).

According to https://pypi.org/project/autopep8/ we are using the latest version 1.5.4.
```
# tox -e fix
fix installed: astroid==2.4.2,autopep8==1.5.4,docutils==0.16,entrypoints==0.3,flake8==3.7.8,flake8-colors==0.1.6,isort==4.3.21,lazy-object-proxy==1.4.3,mccabe==0.6.1,pycodestyle==2.5.0,pyfakefs==4.1.0,pyflakes==2.1.1,pylint==2.3.1,rstcheck==3.3.1,six==1.15.0,toml==0.10.1,wrapt==1.12.1
fix run-test-pre: PYTHONHASHSEED='1782656327'
fix run-test: commands[0] | python --version
Python 3.8.5
fix run-test: commands[1] | autopep8 --max-line-length 100 --exclude '.tox,
.git,
__pycache__,
build,
dist,
*.egg-info,
.cache,
*.pyc,
.eggs,
venv,
frontend,' --in-place --recursive .
```
I have no idea why `autopep8` reorganizes the imports this way (which looks incorrect according to PEP-8).

So what shall we do? Any ideas?

Reply:
Shall we add `# noqa` to every import line that is incorrectly formatted by `autopep8`?

Reply:
I think we should also integrate [isort](https://pypi.org/project/isort/) which could automatically format the import section correctly.

Reply:
Google's [YAPF](https://github.com/google/yapf ) (but perhaps it's too drastic)?

Reply:
And BTW they also mentioned [pep8ify](https://github.com/spulec/pep8ify) as an alternative to autopep8?